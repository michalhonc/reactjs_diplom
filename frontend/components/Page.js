import React, { Component } from 'react';
import styled, { ThemeProvider, createGlobalStyle } from 'styled-components';

import Header from './Header/Header';
import Meta from './Header/Meta';

import theme from './styles/config/config';
import { MenuProvider } from './Menu/MenuContext';
import Menu from './Menu/Menu';
  

const StyledPage = styled.div`
    color: ${props => props.theme.color.black};
    position: relative;
`;

const Inner = styled.div`
    max-width: ${props => props.theme.size.maxWidth};
    margin: 0 auto;
    padding: 2rem;
`;

const GlobalStyle = createGlobalStyle`
    html {
        box-sizing: border-box;
        font-size: 10px;
        font-family: 'Arial CE', Arial, 'Helvetica CE', Helvetica, sans-serif;
        font-weight: normal;
    }

    *, *:before, *:after {
        box-sizing: inherit;
    }

    body {
        padding: 0;
        margin: 0;
        font-size: 1.5rem;
        line-height: 2;
        height: 100%;
        background-color: ${props => props.theme.color.backgroundGray}
    }

    a {
        text-decoration: none;
        color: ${theme.black};
    }

    h1 {
        font-size: 34px;
        line-height: 48px;
        margin: 0;
    }
    h2 {
        font-size: 24px;
        line-height: 32px;
    }
    h3 {
        font-size: 20px;
        line-height: 32px;
    }
    h4 {
        font-size: 16px;
        line-height: 24px;
    }
    h5 {
        font-size: 14px;
        line-height: 24px;
    }
    h6 {
        font-size: 12px;
        line-height: 24px;
    }
`;

class Page extends Component {
    render() { 
        return (
            <ThemeProvider theme={theme}>
                <MenuProvider>
                    <StyledPage>
                        <Meta />
                        <Menu />
                        <Header />
                        <Inner>
                            {this.props.children}
                        </Inner>
                        <GlobalStyle theme={theme} />
                    </StyledPage>
                </MenuProvider>
            </ThemeProvider>
        );
    }
}
 
export default Page;