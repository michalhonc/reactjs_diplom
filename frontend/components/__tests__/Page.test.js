import React from 'react';
import Page from '../Page';
import Index from '../../pages/index';
import renderer from 'react-test-renderer';

it('App render index page', () => {
    const tree = renderer.create(
        <Page>
            <Index />
        </Page>
    );
    expect(tree.toJSON())
        .toMatchSnapshot();
});
