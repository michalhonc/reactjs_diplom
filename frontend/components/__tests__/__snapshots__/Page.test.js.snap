// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render index page 1`] = `
<div
  className="Page__StyledPage-sc-1lc0jd-0 jufmcl"
>
  <header
    className="Header__StyledHeader-sc-6ii4zc-0 byRpXF"
  >
    <h1
      className="Header__Logo-sc-6ii4zc-1 gMQudC"
    >
      <a
        className="Header__StyledAnchor-sc-6ii4zc-2 jkaAbg"
        onClick={[Function]}
      >
        <img
          className="Header__StyledImg-sc-6ii4zc-3 dVCYVW"
          src="/static/icons/s-logo.png"
        />
        Trendy
      </a>
    </h1>
    <nav
      className="Nav__NavStyled-pb472o-0 dcUoYK"
    >
      <button
        className="Button-lh6zao-0 eerQNO"
        onClick={[Function]}
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <svg
          className="icon icon-menu Nav__MenuStyled-pb472o-1 bXSTrH"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="../static/icons/icons.svg#icon-menu"
          />
        </svg>
      </button>
    </nav>
  </header>
  <div
    className="Page__Inner-sc-1lc0jd-1 etDWEt"
  >
    <div
      className="SearchInput__InputWrapper-sc-1nu51op-1 kMySmv"
    >
      <input
        className="SearchInput__Input-sc-1nu51op-0 gnstrE"
        onChange={[Function]}
        placeholder="Hledejte.."
        type="input"
        value=""
      />
      <svg
        className="icon icon-search SearchInput__StyledIcon-sc-1nu51op-2 kQpKlt"
        viewBox="0 0 16 16"
      >
        <use
          xlinkHref="../static/icons/icons.svg#icon-search"
        />
      </svg>
    </div>
    <div
      className="Results__StyledWrapper-sc-1g02b5t-0 DeKtb"
    />
  </div>
</div>
`;
